.github/workflows/integration.yml

```
@@ -0,0 +1,48 @@
+name: integration
+
+on:
+  push:
+    paths-ignore:
+      - 'docs/**'
+      - '*.md'
+  pull_request:
+    paths-ignore:
+      - 'docs/**'
+      - '*.md'
+
+jobs:
+  install-production:
+    runs-on: ${{ matrix.os }}
+    permissions:
+      contents: read
+    
+    strategy:
+      matrix:
+        node-version: [14, 16, 18]
+        os: [ubuntu-latest]
+
+    steps:
+      - uses: actions/checkout@main
+        with:
+          persist-credentials: false
+      
+      - name: Use Node.js
+        uses: actions/setup-node@main
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Install Pnpm
+        run: |
+          npm i -g pnpm
+      
+      - name: Install Production
+        run: |
+          pnpm install --prod
+
+      - name: Run server
+        run: |
+          node integration/server.js &
+
+      - name: Test
+        run: |
+          bash integration/test.sh
```

We should make sure the server started before we try to talk to it:

```suggestion
      - name: Test
        if: ${{ success() }}
        run: |
          bash integration/test.sh
```